### PR TITLE
feat: init doc-plugin-preview

### PR DESCRIPTION
**Pull request by 10Derozan**
### <samp>🤖 Generated by Copilot at cd63119</samp>

This pull request adds a new package `doc-plugin-preview` that provides a plugin for previewing code snippets in the doc pages. It also updates the `doc-core` package to support a new blank page type, fix a bug with the pageType property, and use type-only imports and exports for the shared types. It also refactors and improves the styles and code style of the `doc-core` package.

### <samp>🤖 Generated by Copilot at cd63119</samp>

*  Add a new package doc-plugin-preview that provides a plugin for previewing code snippets in the doc pages (F5-F15, F17-F19)
*  Add a new value blank to the PageType type to support rendering blank pages without navbar or sidebar ([link](https://github.com/web-infra-dev/modern.js/pull/3767/files?diff=unified&w=0#diff-3b9ef78ff832598dcdf880304579f2f3fc95e1dc287ff5a94f1d0e04b162927eL211-R211))
*  Modify the getContentLayout function to support the new blank page type and render the Content component without navbar for it ([link](https://github.com/web-infra-dev/modern.js/pull/3767/files?diff=unified&w=0#diff-ad0dae9c2c9edde59e66297213d95b5b033e379f05ff677cdca9c350ca541357L74-R76))
*  Add a conditional rendering of the Nav component to hide the navbar for the blank page type ([link](https://github.com/web-infra-dev/modern.js/pull/3767/files?diff=unified&w=0#diff-ad0dae9c2c9edde59e66297213d95b5b033e379f05ff677cdca9c350ca541357L124-R132))
*  Use type-only imports and exports to avoid unnecessary code generation and bundling for the shared types ([link](https://github.com/web-infra-dev/modern.js/pull/3767/files?diff=unified&w=0#diff-3b9ef78ff832598dcdf880304579f2f3fc95e1dc287ff5a94f1d0e04b162927eL6-R15))
*  Remove an empty line in the Layout component to improve the code style and readability ([link](https://github.com/web-infra-dev/modern.js/pull/3767/files?diff=unified&w=0#diff-ad0dae9c2c9edde59e66297213d95b5b033e379f05ff677cdca9c350ca541357L116))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/cli/doc-plugin-preview/package.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+  "name": "@modern-js/doc-plugin-preview",
+  "version": "2.19.1",
```
Comment (chenjiahan) [inline packages/cli/doc-plugin-preview/package.json:3]:
```suggestion
  "version": "2.21.1",
```

File: packages/cli/doc-plugin-preview/src/components/demo.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export default function Demo() {
+  // get the id from the pathname
+  const { pathname } = useLocation();
+  const id = pathname.split('/').pop();
+  // get component from virtual-meta
+  const result = demos.flat().find(item => item.id === id);
+  if (result) {
+    return createElement(result.component);
+  } else {
```
Comment (chenjiahan) [inline packages/cli/doc-plugin-preview/src/components/demo.tsx:13]: unnecessary else

File: packages/cli/doc-plugin-preview/package.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    "@arco-design/web-react": "^2.46.0",
+    "@modern-js/module-tools": "workspace:*",
+    "@modern-js/tsconfig": "workspace:*",
+    "@scripts/build": "workspace:*",
+    "@types/mdast": "^3.0.10",
+    "@types/node": "^18.11.17",
+    "@types/react": "^18",
+    "@types/react-dom": "^18",
+    "@types/yaml-front-matter": "^4.1.0",
+    "husky": "^8",
```
Comment (chenjiahan) [inline packages/cli/doc-plugin-preview/package.json:54]: Remove redundant dependencies, such as husky, lint-staged

File: packages/cli/doc-plugin-preview/package.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    "dist/"
+  ],
+  "dependencies": {
+    "@mdx-js/mdx": "2.2.1",
+    "@modern-js/doc-core": "workspace:*",
+    "@modern-js/utils": "workspace:*",
+    "qrcode.react": "^3.1.0"
+  },
+  "devDependencies": {
+    "@arco-design/web-react": "^2.46.0",
```
Comment (chenjiahan) [inline packages/cli/doc-plugin-preview/package.json:45]: @arco-design/web-react should be a dependency

File: packages/cli/doc-plugin-preview/.npmrc [hunk 5]
```diff
@@ -0,0 +1 @@
+strict-peer-dependencies=false
```
Comment (chenjiahan) [inline packages/cli/doc-plugin-preview/.npmrc:1]: Do we need .npmrc?

File: packages/cli/doc-plugin-preview/package.json [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    "@types/react-dom": "^18",
+    "@types/yaml-front-matter": "^4.1.0",
+    "husky": "^8",
+    "lint-staged": "~13.1.0",
+    "mdast-util-mdxjs-esm": "^1.3.0",
+    "prettier": "^2.6.2",
+    "react": ">=17",
+    "react-router-dom": "^6.8.1",
+    "rspack-plugin-virtual-module": "^0.1.4",
+    "typescript": "^4.8.4",
```
Comment (chenjiahan) [inline packages/cli/doc-plugin-preview/package.json:61]: Upgrade to typescript v5

File: packages/cli/doc-plugin-preview/mdx-meta-loader.cjs [hunk 7]
```diff
@@ -0,0 +1,61 @@
+const { join } = require('path');
+const { routeMeta } = require('./dist');
+
+module.exports = async function (source) {
+  const callback = this.async();
+  const { createProcessor } = await import('@mdx-js/mdx');
+  const { visit } = await import('unist-util-visit');
```
Comment (chenjiahan) [inline packages/cli/doc-plugin-preview/mdx-meta-loader.cjs:7]: Can we use `await import` in a .cjs file?

File: packages/cli/doc-plugin-preview/src/locales.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    collapse: 'Collapse Code',
+    className: 'Additional css class',
+    style: 'Additional style',
+    children: 'Children',
+    disabled: 'Whether to disable ',
+    required: 'Required',
+    property: 'Property',
+    description: 'Description',
+    type: 'Type',
+    defaultValue: 'DefaultValue',
```
Comment (chenjiahan) [inline packages/cli/doc-plugin-preview/src/locales.ts:32]:
```suggestion
    defaultValue: 'Default Value',
```

File: packages/cli/doc-plugin-preview/package.json (hunk 4 above)
Comment (10Derozan) [reply packages/cli/doc-plugin-preview/package.json:45]: i have bundled it

File: packages/cli/doc-plugin-preview/package.json (hunk 1 above)
Comment (10Derozan) [reply packages/cli/doc-plugin-preview/package.json:3]: I'll update it after merge  main branch

File: packages/cli/doc-plugin-preview/package.json (hunk 6 above)
Comment (10Derozan) [reply packages/cli/doc-plugin-preview/package.json:61]: ok

File: packages/cli/doc-plugin-preview/mdx-meta-loader.cjs (hunk 7 above)
Comment (10Derozan) [reply packages/cli/doc-plugin-preview/mdx-meta-loader.cjs:7]: yes

File: packages/cli/doc-plugin-preview/.npmrc (hunk 5 above)
Comment (10Derozan) [reply packages/cli/doc-plugin-preview/.npmrc:1]: copy from  other doc plugins

File: packages/cli/doc-plugin-preview/src/components/demo.tsx (hunk 2 above)
Comment (10Derozan) [reply packages/cli/doc-plugin-preview/src/components/demo.tsx:13]: maybe i need add a fallback case

File: packages/cli/doc-plugin-preview/src/locales.ts (hunk 8 above)
Comment (10Derozan) [reply packages/cli/doc-plugin-preview/src/locales.ts:32]: fix

**Review by sanyuan0704 — APPROVED** [review]

**Pull request merged**